ocaml/parsing/parser.mly

```diff
@@ -303,26 +274,6 @@ let rec mktailpat nilloc = let open Location in function
 let mkstrexp e attrs =
   { pstr_desc = Pstr_eval (e, attrs); pstr_loc = e.pexp_loc }
 
-<<<<<<< HEAD
-let mkexp_desc_type_constraint e t =
-  match t with
-  | Pconstraint t -> Pexp_constraint(e, t)
-  | Pcoerce(t1, t2)  -> Pexp_coerce(e, t1, t2)
-
-let mkexp_type_constraint ?(ghost=false) ~loc e t =
-  let desc = mkexp_desc_type_constraint e t in
-  if ghost then ghexp ~loc desc
-  else mkexp ~loc desc
-||||||| a198127529
-let mkexp_type_constraint ?(ghost=false) ~loc e t =
-  let desc =
-    match t with
-  | N_ary.Pconstraint t -> Pexp_constraint(e, t)
-  | N_ary.Pcoerce(t1, t2)  -> Pexp_coerce(e, t1, t2)
-  in
-  if ghost then ghexp ~loc desc
-  else mkexp ~loc desc
-=======
```

Comment:
(I'm attempting to leave the comment on the portion below the `=======` but GitHub isn't cooperating.)

I find this implementation to be pretty sad. `Pcoerce` just drops `~modes`? Sure, I think it's always empty here, but the code structure doesn't make that clear. Probably we should move the modes to the payload of `Pconstraint` &mdash; that will simplify some other things too. Let's CR this.

(I know you didn't write this code.)